Array.prototype.to spliced() (#2901)
Codecademy/docs: content/javascript/concepts/arrays/terms/toSpliced/toSpliced.md

---
 Title: '.toSpliced()'
Description: 'Returns a new array with deleted, replaced, or inserted values at the given index.'
Subjects:
  - 'Web development'
Tags:
  - 'Arrays'
  - 'Methods'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

> 
> ## .toSpliced()

Reviewer: ```suggestion
```

> **`.toSpliced()`** is a method that modifies multiple array elements in one go. It usually has a start point (the first element to be modified) and an end point (the last element to be modified). `.toSpliced()` alters array elements in the following ways: 

Reviewer: ```suggestion
**`.toSpliced()`** is a method that modifies multiple array elements in one go. It usually has a start point (the first element to be modified) and an end point (the last element to be modified). `.toSpliced()` can make the following changes to an array: 
```

- Delete element(s)
- Replace element(s)
- Insert element(s)

> **Note:** `.toSpliced()` returns the copy of an array with the given modifications. It does not mutate the parent array.

> ## Syntax

Reviewer: add an empty line after the syntax header

```pseudo
array.toSpliced(startIndex, count, elementN)
```

`.toSpliced` takes the following parameters:

> - `startIndex`: the location of the first element to be altered.
> 
> - `count`: the number of elements that should be modified, beginning from `startIndex`
> 
> - `elementN`: the updated value(s) for the array element(s). To insert multiple values, seperate each value with a comma. 

Reviewer: ```suggestion
- `startIndex`: The location of the first element to be altered.
- `count`: The number of elements that should be modified, beginning from `startIndex`.
- `elementN`: The updated value(s) for the array element(s). To insert multiple values, seperate each value with a comma. 
```

> ## Examples

Reviewer: Add an empty line after examples header

> #### Delete array elements with `.toSpliced`

Reviewer: ```suggestion
#### Delete Array Elements With `.toSpliced`
```

```js
const colors = ['red', 'yellow', 'blue', 'orange', 'green', 'purple'];

> // Removing  red, yellow, and blue. Start and index 0 and delete 3 elements

Author: I agree with this suggestion. But it raises a new issue.

 If .toSpliced() extracts elements from one array and places them in another, then maybe it creates a new array - rather than modifies the original array. I'd say this logic applies to all the other copying methods as well. 

I'm committing the changes based on your current suggestions now. But I'll be happy to make more changes.

const primaryColors = colors.toSpliced(0, 3);

console.log(primaryColors);

> //Prints 'red', 'yellow', 'blue'

Reviewer: ```suggestion
// Prints 'red', 'yellow', 'blue'
```

console.log(colors);
// Prints 'red', 'yellow', 'blue', 'orange', 'green', 'purple'
```

> #### Replace array elements using `.toSpliced()`

Reviewer: ```suggestion
#### Replace Array Elements Using `.toSpliced()`
```

```js
const colors = ['red', 'yellow', 'blue', 'orange', 'green', 'purple'];

//Replacing red, green and yellow. Start at index 0, and replace three items. 

const tertiaryColors = colors.toSpliced(0, 3, 'blue-green', 'red-violet', 'yellow-orange');

console.log(tertiaryColors);
//Prints 'blue-green', 'red-violet', 'yellow-orange', 'orange', 'green', 'purple'

console.log(colors);
// Prints 'red', 'yellow', 'blue', 'orange', 'green', 'purple'
```

> #### Insert new items into an array using `.toSpliced`

Reviewer: ```suggestion
#### Insert New Items Into an Array Using `.toSpliced`
```
